lib/src/Uncertainty/Distribution/DiscreteCompoundDistribution.cxx

```diff
@@ -0,0 +1,289 @@
+  if (!compound_distribution.isIntegral()) throw InvalidArgumentException(HERE) << "Error: the compound distribution must be integer-valued.";
+
+  baseDistribution_ = base_distribution;
+  compoundDistribution_ = compound_distribution;
+
+ 
+ // Distribution of the sum
+
+  createIntegralCompound();
+}
+
+/* Constructs PDF */
+
+/* Comparison operator */
+Bool DiscreteCompoundDistribution::operator ==(const DiscreteCompoundDistribution & other) const
+{
+  if (this == &other) return true;
+  return baseDistribution_ == other.baseDistribution_ && compoundDistribution_ == other.compoundDistribution_;
+}
+
+Bool DiscreteCompoundDistribution::equals(const DistributionImplementation & other) const
+{
+  const DiscreteCompoundDistribution* p_other = dynamic_cast<const DiscreteCompoundDistribution*>(&other);
+  return p_other && (*this == *p_other);
+}
+
+/* String converter */
+String DiscreteCompoundDistribution::__repr__() const
+{
+  OSS oss(true);
+  oss << "class=" << DiscreteCompoundDistribution::GetClassName()
+      << " name=" << getName();
+      //<< " dimension=" << getDimension()
+      //<< " p=" << p_;
+  return oss;
+}
+
+String DiscreteCompoundDistribution::__str__(const String & ) const
+{
+  OSS oss(false);
+  oss << getClassName() << "Base distribution " << baseDistribution_ << " Compound distribution" << compoundDistribution_;
+  return oss;
+}
+
+/* Virtual constructor */
+DiscreteCompoundDistribution * DiscreteCompoundDistribution::clone() const
+{
+  return new DiscreteCompoundDistribution(*this);
+}
+
+/* Compute the numerical range of the distribution given the parameters values */
+void DiscreteCompoundDistribution::computeRange()
+{
+  const Point lowerBound(1, 0.0);
+  const Point upperBound(1, computeIntegerUpperBound());
+  const Interval::BoolCollection finiteLowerBound(1, true);
+  const Interval::BoolCollection finiteUpperBound(1, false);
+  setRange(Interval(lowerBound, upperBound, finiteLowerBound, finiteUpperBound));
+}
+
+/* Get one realization of the distribution */
+Point DiscreteCompoundDistribution::getRealization() const
+{
+  return distribution_.getRealization();
+}
+
+
+/* All values of the PDF */
+
+Point DiscreteCompoundDistribution::computeProbabilities(const UnsignedInteger m, const Scalar epsilon) const // FIXME
+{
+
+  if (epsilon <= 0) throw InvalidArgumentException(HERE) << "Error parameter must be positive.";
+
+  Scalar r = std::pow(epsilon, 1.0/m);
+
+
+  FFT fft;
+  FFT::ComplexCollection collection = FFT::ComplexCollection (m);
+  for(UnsignedInteger i=0;i<m;i++)
+  {
+    collection[i] = computeGeneratingFunction(r * std::exp(2.0 * Complex(0.0,1.0) * (M_PI * i / m)));
+  }
+
+  collection = fft.transform(collection);
+
+  Point probabilities(m);
+
+  for(UnsignedInteger i=0;i<m;i++)
+  {
+	probabilities[i] = std::max(0.0, collection[i].real() / m / std::pow(r,i));
```

Comment:
There's a small compilation error with visual studio here, it should be
`std::pow(r, (Scalar)i)` instead of `std::pow(r,i)`